Move embargo from common to openedx/core/djangoapps.

doctoryes opened the pull request:
@robrap @andy-armstrong @nasthagiri @jcdyer Here's my remaining `clean_platform` PR for your review. It doesn't use the relative pathing with the `embargo` Django app that we've decided upon - FYI.

robrap commented:
@doctoryes Looks good.  

Questions:
1. Any reason not to use relative pathing?  Do we think it isn't a big deal either way, and we should let developers decide.  Or should we have a documented best practice and have people use it?
2. Did you check migrations?  What were the results?

Thanks!

doctoryes commented:
jenkins run js

doctoryes commented:
@robrap :
1) I changed all pathing to relative. It's not really a big deal. The benefits I see are a) making it clearer which imports are from within the module itself *and* b) making it easier to move to a module into a separate repo and out of edx-platform. At most, the best practice would be to prefer relative pathing.
2) I checked migrations - there are no migrations generated when I run `makemigrations`. Does that cover your question?

@andy-armstrong @nasthagiri I squashed this PR and it's ready for a final review.

robrap commented:
@doctoryes: Thanks for the import clean-up. 👍  on the code.  I did not do any testing.

doctoryes commented:
@andy-armstrong I did not do any additional testing of the embargo functionality. Do you recommend that I do so? I guess it *is* law-sensitive functionality.

doctoryes commented:
OK - I've now successfully tested the feature on a sandbox:
http://embargo-test.sandbox.edx.org
and I've written up instructions on how to test the embargo feature:
https://openedx.atlassian.net/wiki/display/PLAT/How+To+Test+the+Course+Embargo+Feature

I also have two thumbs - so I'm merging this PR.

andy-armstrong commented:
@doctoryes thanks so much for going the extra mile to write up the instructions. I really appreciate it!
